appengine/accessible/levelmanager.component.js

```diff
@@ -0,0 +1,207 @@
+  var topBlock = blocklyApp.workspace.topBlocks_[0];
+  var errorMessage = 'Not quite! Try again!';
+
+  //we should only report the earliest problem with their code
+  var errorMessageChanged = false;
+
+  expectedBassLine.setFromChordsAndDurations(
+    musicGame.gameManager.expectedBassLines[level]);
+  correct = correct && musicPlayer.doesBassLineEqual(expectedBassLine);
+  if (!correct && !errorMessageChanged){
+    errorMessage = 'Not quite! Are you playing the right note?'
+    errorMessageChanged = true;
+  }
+
+  if (level != 3) {
+    correct = correct && blocklyApp.workspace.topBlocks_.length == 1;
+  } else {
+    //if there are two topblocks that aren't connected, the error message should be the error message on line 112
+    correct = correct && (blocklyApp.workspace.topBlocks_.length == 1 
+        || blocklyApp.workspace.topBlocks_.length == 2);
+  }
+  if (!correct && !errorMessageChanged) {
+    errorMessage = 'Not quite! Are you playing the right number of blocks?'
+    errorMessageChanged = true;
+  }
+
+  correct = correct && topBlock && topBlock.type == 
+      musicGame.gameManager.expectedBlockType[level];
+  if (!correct && !errorMessageChanged) {
+    errorMessage = 'Not quite! Are you playing the right block?'
+    errorMessageChanged = true;
+  }
+
+  if (level == 3) {
+    var connection = topBlock.nextConnection.targetConnection;
+    correct = correct && connection && connection.sourceBlock_ && 
+        connection.sourceBlock_.type == musicGame.gameManager.expectedBlockType[level];
+    if (!correct && !errorMessageChanged) {
+      errorMessage = 'Not quite! Are your blocks connected?'
+      errorMessageChanged = true;
+    }
+  }
+  if (correct) {
+    alert('Good job! You completed the level!');
+    musicGame.gameManager.maxLevelAllowed = Math.min(musicGame.gameManager.maxLevelAllowed + 1, 3);
+  } else {
+    alert(errorMessage);
+  }
+}
+
+musicGame.LevelManagerView = ng.core
+  .Component({
+    selector: 'levelview',
+    template: `
+    <h2>Page Layout</h2>
+    <p>Welcome to the Blockly Music Game! On this page is a music puzzle game
+    you will complete using AccessibleBlockly. There are three levels to this
+    game.</p>
+    <p>This page has several instruction sections followed by the game area.
+    All sections have a level 2 heading associated with them. The
+    ‘Screen Reader Settings and Compatilibity’ section (directly after this
+      section) describes the best screen reader/browser combinations to use for
+    AccessibleBlockly and what settings to use. The ‘Navigation’ section
+    describes how to use your keyboard to navigate in the AccessibleBlockly game
+     area. The ‘Level Goal’ section tells you the objective of the level you are
+      on. You are currently on level {{level}}. The ‘Hints’ section will be available if
+       the level has any hints associated with it. Any extra tips or reminders
+       will be in the ‘Hints’ section.</p>
```

Comment:
This is due to trying to keep the 180 character limit. Some spaces ended up being at the front. I can change that if you'd like.